src/System.Net.Sockets/src/System/Net/Sockets/IOControlKeepAlive.cs

```diff
@@ -0,0 +1,98 @@
+            {
+                return !s_supportsKeepAliveViaSocketOption;
+            }
+        }
+
+        public static int Get(SafeSocketHandle handle, SocketOptionName optionName)
+        {
+            IOControlKeepAlive ioControlKeepAlive;
+            s_socketKeepAliveTable.TryGetValue(handle, out ioControlKeepAlive);
+            return optionName == SocketOptionName.TcpKeepAliveTime ?
+                MillisecondsToSeconds(ioControlKeepAlive?._timeMs ?? WindowsDefaultTimeMs) :
+                MillisecondsToSeconds(ioControlKeepAlive?._intervalMs ?? WindowsDefaultIntervalMs);
+        }
+
+        public static SocketError Set(SafeSocketHandle handle, SocketOptionName optionName, int optionValueSeconds)
+        {
+            IOControlKeepAlive ioControlKeepAlive = s_socketKeepAliveTable.GetOrCreateValue(handle);
+            if (optionName == SocketOptionName.TcpKeepAliveTime)
+                ioControlKeepAlive._timeMs = SecondsToMilliseconds(optionValueSeconds);
+            else
+                ioControlKeepAlive._intervalMs = SecondsToMilliseconds(optionValueSeconds);
+
+            byte[] buffer = s_keepAliveValuesBuffer ?? (s_keepAliveValuesBuffer = new byte[3 * sizeof(uint)]);
+            ioControlKeepAlive.Fill(buffer);
+            int realOptionLength = 0;            
+            return SocketPal.WindowsIoctl(handle, unchecked((int)IOControlCode.KeepAliveValues), buffer, null, out realOptionLength);
+        }
+
+        private static bool SupportsKeepAliveViaSocketOption()
+        {
+            using (Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                int time = MillisecondsToSeconds(WindowsDefaultTimeMs);
+                SocketError timeErrCode = Interop.Winsock.setsockopt(
+                    socket.SafeHandle,
+                    SocketOptionLevel.Tcp,
+                    SocketOptionName.TcpKeepAliveTime,
+                    ref time,
+                    sizeof(int));
+                int interval = MillisecondsToSeconds(WindowsDefaultIntervalMs);
+                SocketError intervalErrCode = Interop.Winsock.setsockopt(
+                    socket.SafeHandle,
+                    SocketOptionLevel.Tcp,
+                    SocketOptionName.TcpKeepAliveInterval,
+                    ref interval,
+                    sizeof(int));
+                return timeErrCode == SocketError.Success && intervalErrCode == SocketError.Success;
+            }
+        }
+
+        private static int MillisecondsToSeconds(uint milliseconds)
+        {
+            return (int)(milliseconds / 1000u);
+        }
+
+        private static uint SecondsToMilliseconds(int seconds)
+        {
+            return (uint)seconds * 1000u;
+        }
+
+        private void Fill(byte[] buffer)
+        {
+            Debug.Assert(buffer != null);
+            Debug.Assert(buffer.Length == 3 * sizeof(uint));
+
+            const int uintSize = sizeof(uint);
```

Comment:
There's no need to store this into a const.  You can just use `sizeof(uint)` at the call site.